include/deal.II/base/mpi.h

```diff
@@ -952,6 +967,28 @@ namespace Utilities
 #  endif
     }
 
+
+#  ifdef DEAL_II_WITH_MPI
+    template <class Iterator, typename Number>
+    std::pair<Number, typename numbers::NumberTraits<Number>::real_type>
+    mean_std(const Iterator begin, const Iterator end, const MPI_Comm &comm)
+    {
+      using Std        = typename numbers::NumberTraits<Number>::real_type;
+      const Number sum = std::accumulate(begin, end, Number(0.));
+
+      const auto size =
+        static_cast<Std>(Utilities::MPI::sum(std::distance(begin, end), comm));
```

Comment:
why do you need the cast? The size is still an integer type.

Reply:
`std::complex<double>/size` ain't gonna work.

Reply:
Yes, I remember. But then leave `size` as an integral type and only do the cast where you do the division.

Reply:
but that does not influence the result, does it? I would also need to cast twice in this case.

Reply:
Correct -- same result. 

I'm thinking about this from the perspective of readability of code. Because you had made the cast and declared the variable with an odd type, I stumbled over it as I didn't understand the need for it. A reader in a year would likely do the same -- "why the heck does the variable not have an integral type if it stores an integer?" On the other hand, if the cast happened in the place where you do the division, I know that it must have something to do with the division you're doing, and I can specifically look up whether `std::complex` has a division operator with integers (which it doesn't).